.github/scripts/matrix.py

```diff
@@ -64,6 +71,7 @@ def to_dict(self) -> Dict[str, Union[str, int]]:
 class BuildConfig:
     arch: Arch
     toolchain: Toolchain
+    bpf_toolchain: Toolchain = Toolchain(compiler=Compiler.LLVM, version=DEFAULT_LLVM_VERSION)
```

Comment:
this is what seems to cause the `set_matrix` step failure. 

The error is:
```
Traceback (most recent call last):
  File "/tmp/work/bpf/bpf/.github/scripts/matrix.py", line 70, in <module>
    @dataclasses.dataclass
     ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/dataclasses.py", line 1268, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/usr/lib/python3.12/dataclasses.py", line 12[5](https://github.com/kernel-patches/bpf/actions/runs/10881506538/job/30190585441?pr=7737#step:3:6)8, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/dataclasses.py", line 994, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/dataclasses.py", line [8](https://github.com/kernel-patches/bpf/actions/runs/10881506538/job/30190585441?pr=7737#step:3:9)52, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class '__main__.Toolchain'> for field bpf_toolchain is not allowed: use default_factory
```

From https://stackoverflow.com/a/75677129

it seems https://gist.github.com/chantra/b4ac937f0f0e909d238bb03f7aacae49 would work.